plugins/modules/xdg_mime.py

```diff
@@ -0,0 +1,156 @@
+      - The desktop file must be installed in the system.
+        If the desktop file is not installed, the module
+        will not fail, but the handler will not be set either.
+    type: str
+    required: true
+notes:
+  - This module is a thin wrapper around xdg-mime tool.
+  - See man xdg-mime(1) for more details.
+seealso:
+  - name: C(xdg-mime) command manual page
+    description: Manual page for the command.
+    link: https://portland.freedesktop.org/doc/xdg-mime.html
+  - name: xdg-utils Documentation
+    description: Reference documentation for xdg-utils.
+    link: https://www.freedesktop.org/wiki/Software/xdg-utils/
+"""
+
+EXAMPLES = r"""
+- name: Set Chrome as the default handler for HTTPS
+  community.general.xdg_mime:
+    mime_type: x-scheme-handler/https
+    handler: google-chrome.desktop
+  register: result
+- name: Set Chrome as the default handler for HTTP/HTTPS
+  community.general.xdg_mime:
+    mime_type:
+      - x-scheme-handler/http
+      - x-scheme-handler/https
+    handler: google-chrome.desktop
+  register: result
+  """
+
+RETURN = r"""
+after_handlers:
+  description:
+    - List of handlers set by the module.
+  returned: success
+  type: list
+  elements: str
+  sample:
+    - google-chrome.desktop
+    - google-chrome.desktop
+before_handlers:
+  description:
+    - List of handlers currently set.
+  returned: success
+  type: list
+  elements: str
+  sample:
+    - google-chrome.desktop
+    - firefox.desktop
+version:
+  description: Version of xdg-mime.
+  type: str
+  returned: always
+  sample: "1.2.1"
+"""
+
+from ansible_collections.community.general.plugins.module_utils.module_helper import ModuleHelper
+from ansible_collections.community.general.plugins.module_utils.xdg_mime import xdg_mime_runner, xdg_mime_get
+
+
+class XdgMime(ModuleHelper):
+    output_params = ['before_handlers', 'after_handlers']
+    diff_params = ['before_handlers', 'after_handlers']
+    module = dict(
+        argument_spec=dict(
+            mime_type=dict(type='list', elements='str', required=True),
+            handler=dict(type='str', required=True),
+        ),
+        supports_check_mode=True,
+    )
+    use_old_vardict = False
+
+    def __init_module__(self):
+        self.runner = xdg_mime_runner(self.module, check_rc=True)
+
+        with self.runner("version") as ctx:
+            rc, out, err = ctx.run()
+            self.vars.version = out.replace("xdg-mime ", "").strip()
+
+        self.vars.before_handlers = []
+        for mime in self.vars.mime_type:
+            handler_value = xdg_mime_get(self.runner, mime)
+            if not handler_value:
+                handler_value = ''
+            self.vars.before_handlers.append(handler_value)
+
+    def __run__(self):
+        check_mode_return = (0, 'Module executed in check mode', '')
+
+        if not self.vars.handler.endswith(".desktop"):
+            self.do_raise(msg="Handler must be a .desktop file")
```

Comment:
I added the check inside the code instead let the xdg-mime report, because I'm mocking the result with check mode, but the problem didn't happen when using check mode since the main part (ctx.run) wouldn't run during check mode.

Reply:
Ok, that is a good call. 

But if I may suggest, do this earlier, in `__init_module__()`, so that you pick the error before even running `xdg-mime query`. Naturally you will need to remove the query call from the unit test, if you do that.